src/main/java/decodes/tsdb/RecordRangeHandle.java

```diff
@@ -20,92 +20,113 @@
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.Iterator;
+import java.util.List;
+
+import org.opendcs.tsdb.FailedTaskListEntry;
+import org.opendcs.tsdb.TaskListEntry;
 
 import decodes.sql.DbKey;
 
 public class RecordRangeHandle
 {
 	private DbKey appId;
-	private ArrayList<Integer> recnums;
-	private HashSet<Integer> failed_recnums;
+	private ArrayList<TaskListEntry> recNums;
+	private HashSet<FailedTaskListEntry> failedRecNums;
 
 	public RecordRangeHandle(DbKey appId)
 	{
 		this.appId = appId;
-		recnums = new ArrayList<Integer>();
-		failed_recnums = new HashSet<Integer>();
+		recNums = new ArrayList<>();
+		failedRecNums = new HashSet<>();
 	}
 
 	/** @return the application ID */
-	public DbKey getAppId() { return appId; }
+	public DbKey getAppId()
+	{
+		return appId;
+	}
 
 	/**
 	 * Adds a record number.
-	 * @param recnum the record number
+	 * @param entry the record number
 	 */
-	public void addRecNum(int recordNumber)
+	public void addRecNum(TaskListEntry entry)
 	{
-		recnums.add(recordNumber);
+		recNums.add(entry);
 	}
 
 	/**
 	 * Returns a string containing comma-separated integer record numbers.
 	 * The returned integers are also removed from the list.
```

Comment:
```suggestion
	 * removes and returns a list of TaskListEntry 
```
consider calling this method:  RemoveEntries(max)

Reply:
That's a good point on the naming, I wouldn't expected removal on the current name alone.